### DISCUSSION TITLE
NFT Image Not Displayed on Etherscan and MetaMask

### DISCUSSION THREAD

**Discussion by meitedaf** [Cyfrin/foundry-full-course-cu#2112] [Q&A] [answered]
**Course Section:** Section 2: Develop an NFTs Collection

**Problem Description:**
I successfully deployed my NFT contract and minted an NFT. Here is the contract address on Sepolia: [Contract Address](https://sepolia.etherscan.io/token/0x4d6d1dfa0e882f03bdafb3db12f6baf1d2fbe00f#readContract)

Using the `readContract` function, I queried the `tokenURI` and received the following IPFS link: `ipfs://QmUGveXhXxrNUn28Le3aKygYsFq3zkF9WvwasjaynYawWm`.

When I open the link, it correctly displays the JSON metadata file:
```json
{
    "name": "SHIBA",
    "description": "An adorable SHIBA pup!",
    "image": "ipfs://QmYx6GsYAKnNzZ9A6NvEKV9nf1VaDzJrqDR23Y8YSkebLU",
    "attributes": [
        {
            "trait_type": "cuteness",
            "value": "100"
        }
    ]
}
```
The image link in the JSON file also correctly shows the picture of the SHIBA pup.

However, the image does not display on Sepolia Etherscan or in MetaMask. Both platforms fail to show the SHIBA image associated with the NFT.

**Additional Information:**

	•  Contract Address: [0x4d6d1dFA0E882f03bDAFB3DB12F6BAf1d2FBe00F](https://sepolia.etherscan.io/token/0x4d6d1dfa0e882f03bdafb3db12f6baf1d2fbe00f#readContract)
	•  Metadata URI: ipfs://QmUGveXhXxrNUn28Le3aKygYsFq3zkF9WvwasjaynYawWm
	•  Image URI: ipfs://QmYx6GsYAKnNzZ9A6NvEKV9nf1VaDzJrqDR23Y8YSkebLU
	
I've looked at other similar issues, but the solution doesn't work for me.

Please provide guidance on how to resolve this issue so that the NFT image is correctly displayed on Etherscan and MetaMask.

Thank you.

**Comment by meitedaf** [comment]
## Another Issue:
I minted the same NFT file twice in the same contract using the metadata URI: `ipfs://QmUGveXhXxrNUn28Le3aKygYsFq3zkF9WvwasjaynYawWm`
Both mint operations were successful, resulting in two tokenId 0 and 1, respectively. However, they essentially represent the same NFT. Is this a bug?

**Reply by EngrPips** [reply, marked as answer]
How do you mean `they essentially represent the same NFT` and you can view your `NFT` on opensea, I don't know that sepolia etherscan also displays an `NFT` image. 

You can also go to `privacy & security` settings in your Metamask and check every setting under `privacy & security` that deals with displaying an `NFT` image and enable them then re-import your `NFT`. Let's see how it plays out

**Reply by meitedaf** [reply]
Thank you very much, the method of enabling the settings for `displaying an NFT image and then re-importing the NFT` worked successfully. I can now see the NFT image in MetaMask.

By saying `they essentially represent the same NFT`, I mean that I can use the same metadata URI to mint two or even more NFTs (as shown in the image below).
<img width="340" alt="截屏2024-07-26 15 28 25" src="https://github.com/user-attachments/assets/e84124b2-44b9-4f29-8f56-9861e9072bf1">

Regarding `Sepolia Etherscan also displays an NFT image`, you can refer to my screenshot. Currently, OpenSea and Sepolia Etherscan still do not display the NFT image, but it is worth celebrating that it has successfully appeared in MetaMask!
<img width="1397" alt="截屏2024-07-26 15 28 53" src="https://github.com/user-attachments/assets/88a0095c-6a3a-4de3-8255-7b3acda1fe22">
<img width="927" alt="截屏2024-07-26 15 29 07" src="https://github.com/user-attachments/assets/8e7578a6-0f0b-42f9-87fd-88e07e94df5f">

**Reply by EngrPips** [reply]
I am glad you at least have seen your `NFT` in Metamask. It is weird that Opensea didn't display the image as it used to be the go-to place to see our `NFT,` lol and yeah, the `NFT` for the collection you are minting has the same metadata URI because that is what is in our code but still it is a separate token as you can see that their tokenId is different.

**Reply by meitedaf** [reply]
Thank you! 
Your explanation really helps!

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.

**Reply by WuJingweiPayton** [reply]
@EngrPips 
Hi, Pips! NFT image not visible
weird, I successfully have NFT image visible in Sepolia TestNet, but I do not have visibility in Anvil TestNet.
not happy

![image](https://github.com/user-attachments/assets/9a8d5963-6062-4ecc-b47b-8bf31ae1a87b)
![image](https://github.com/user-attachments/assets/c8bb1e25-ae89-4f49-a33c-60874a429150)

**Reply by EngrPips** [reply]
That is tough, Don't let that slow you down anyways.

**Reply by gregarican** [reply]
Just stumbled across this thread, and found it helpful for me as well. Deployed a smart contract on the Polygon Amoy testnet, then minted a few NFT's. The metadata file pointed to the correct IPFS image, and the tokenURI function pointed to the correct IPFS metadata file. Can pull up those files easily using a web browser. Only after deleting the MetaMask NFT and re-importing it would the image properly appear as expected in my wallet. Curious if it's either some caching issue or else the fact that this is a testnet. The Polygon Amoy block explorer still displays just an image placeholder. But I will move on and consider this some odd bug.

**Reply by EngrPips** [reply]
> Just stumbled across this thread, and found it helpful for me as well. Deployed a smart contract on the Polygon Amoy testnet, then minted a few NFT's. The metadata file pointed to the correct IPFS image, and the tokenURI function pointed to the correct IPFS metadata file. Can pull up those files easily using a web browser. Only after deleting the MetaMask NFT and re-importing it would the image properly appear as expected in my wallet. Curious if it's either some caching issue or else the fact that this is a testnet. The Polygon Amoy block explorer still displays just an image placeholder. But I will move on and consider this some odd bug.

Moving makes sense to me, as getting stuck on such a thing when taking a course doesn't seem like the correct approach. You may come back to figure it out.
